### PR TITLE
Update API from v0.1 to v1.0

### LINKED ISSUE #991: Move API path from v0.1 to v1.0

### PR DESCRIPTION
**Pull request by cliveseldon**
Fixes #991 

 * Updates engine REST controller
 * Update python wrapper
 * update openapi

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **cliveseldon**

If they are not already assigned, you can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | dff116e8c450c12acce5638c323ec3b5c2529e47 | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | dff116e8c450c12acce5638c323ec3b5c2529e47 | [link]() | `/test lint`
pr-build | dff116e8c450c12acce5638c323ec3b5c2529e47 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: python/tests/test_storage.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 import seldon_core
 from minio import Minio, error
 import unittest.mock as mock
-from utils import skipif_gcs_missing
+from .utils import skipif_gcs_missing
```
Comment (ryandawsonuk) [inline python/tests/test_storage.py]: Presumably this is a slip and is also the reason linting is failing?

Comment (ryandawsonuk) [reply python/tests/test_storage.py]: So it wasn't that one, it was a different file - https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1103/1.log#L44

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1103#pullrequestreview-329973879" title="Approved">ryandawsonuk</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [ryandawsonuk]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1103#issuecomment-564116885" title="Approved">ryandawsonuk</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [ryandawsonuk]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Tue Dec 10 16:32:29 UTC 2019
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1103/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1103 --build=4

**Comment by seldondev** [comment]
Tue Dec 10 16:32:33 UTC 2019
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1103/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1103 --build=3

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | d7ec5f0c8aa0a89c9166c0c5eced925560c1e97d | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Wed Dec 11 10:23:18 UTC 2019
The logs for [pr-build] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1103/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1103 --build=5

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | d7ec5f0c8aa0a89c9166c0c5eced925560c1e97d | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by ryandawsonuk** [comment]
@cliveseldon the error from the python tests is
```
Traceback:
tests/test_storage.py:23: in <module>
    from .utils import skipif_gcs_missing
E   ImportError: attempted relative import with no known parent package
_____________________ ERROR collecting tests/test_utils.py _____________________
ImportError while importing test module '/workspace/source/python/tests/test_utils.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/test_utils.py:11: in <module>
    from .utils import skipif_tf_missing
E   ImportError: attempted relative import with no known parent package
```

**Comment by ryandawsonuk** [comment]
That's from the jx server but I also get the same error when I run from this branch locally.

**Comment by ryandawsonuk** [comment]
Presumably this is about which [directory the tests are invoked from](https://realpython.com/absolute-vs-relative-python-imports/). I'm doing `make test` from the python directory. I guess the path is resolving as relative to the `/python` directory rather than `/python/tests` (which is where it needs to point).

**Comment by ryandawsonuk** [comment]
Actually the problem is presumably captured by https://napuzba.com/a/import-error-relative-no-parent/p2

**Comment by ryandawsonuk** [comment]
Yeah it works if I put an empty __init__.py in `/python/tests`

**Comment by seldondev** [comment]
New changes are detected. LGTM label has been removed.

**Comment by seldondev** [comment]
Wed Dec 11 11:25:10 UTC 2019
The logs for [lint] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1103/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1103 --build=7

**Comment by seldondev** [comment]
Wed Dec 11 11:25:12 UTC 2019
The logs for [pr-build] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1103/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1103 --build=6

**Pull request merged**